### ISSUE TITLE
python script triggered by any/random events

### ISSUE THREAD

**Issue by kofec** [domoticz/domoticz#3092]
My version:
```
2019-03-04 19:41:59.050 Status: Domoticz V4.10531 (c)2012-2019 GizMoCuz
2019-03-04 19:41:59.051 Status: Build Hash: bf19d8bb, Date: 2019-02-27 09:04:05
2019-03-04 19:41:59.052 Status: Startup Path: /usr/share/domoticz/
2019-03-04 19:41:59.153 Status: EventSystem: reset all events...
2019-03-04 19:41:59.385 Status: PluginSystem: Started, Python version '3.7.2'.
```
I have created a simple script to test "time" and "event" python scripts. 

```
root@Xiaomi3G:/etc/domoticz/scripts/python# cat script_device_Przelacznik.py
import DomoticzEvents as DEVS
DEVS.Log("Python: Changed: " + DEVS.changed_device.Describe())

DEVS.Log("DEVS.changed_device_name: " + str(DEVS.changed_device_name))
DEVS.Log("DEVS.changed_device.name: " + "ID: " + str(DEVS.changed_device.id) + " Name: " + str(DEVS.changed_device.name))

DEVS.Log("Python: Number of user_variables: " + str(len(DEVS.user_variables)))

# All user_variables are treated as strings, convert as necessary
for key, value in DEVS.user_variables.items():
    DEVS.Log("Python: User-variable '{0}' has value: {1}".format(key, value))

DEVS.Log("DIR(dir(DEVS.changed_device)): " + str(dir(DEVS.changed_device)))
DEVS.Log("DIR(DEVS): " + str(dir(DEVS)))
for device in DEVS.Devices:
   if device:
       DEVS.Log("device: " + str(device) + " Id: " + str(DEVS.Devices[device].id) + " n_value_string: " + str(
           DEVS.Devices[device].n_value_string) + " s_value: " + str(
           DEVS.Devices[device].s_value) + " Last updated: " +
                str(DEVS.Devices[device].last_update_string) + " All typs:" + str(
           DEVS.Devices[device].sub_type) + " " + str(DEVS.Devices[device].switch_type) + " " + str(
           DEVS.Devices[device].type))

root@Xiaomi3G:/etc/domoticz/scripts/python#
```

I have added one dummy switch to test it. I noticed that this scrept was executed many time by any other devices like CPU usage or Memory usage by also by other devices.

Regarding the issue and question 1:
https://github.com/domoticz/domoticz/issues/2097
The variable changed_device and changed_device_name are not present in "time" script only in action right ?
@teeminus 

```
2019-03-04 19:39:14.850  Python: Changed: ID: 5 Name: HDD /overlay, Type: 243, subType: 6, switchType: 0, s_value: 19.42, n_value: 0, n_value_string: 19.42, last_update_string: 2019-03-04 19:39:14
domoticz: DEVS.changed_device_name: HDD /overlay
2019-03-04 19:39:14.851  DEVS.changed_device_name: HDD /overlay
domoticz: DEVS.changed_device.name: ID: 5 Name: HDD /overlay
2019-03-04 19:39:14.852  DEVS.changed_device.name: ID: 5 Name: HDD /overlay
domoticz: Python: Number of user_variables: 0
2019-03-04 19:39:14.853  Python: Number of user_variables: 0
domoticz: DIR(dir(DEVS.changed_device)): ['Describe', '__class__', '__delattr__', '__dir__', '__doc__', '__eq__', '__format__', '__ge__', '__getattribute__', '__gt__', '__hash__', '__init__', '__init_subclass__', '__le__', '__lt__', '__ne__', '__new__', '__reduce__', '__reduce_ex__', '__repr__', '__setattr__', '__sizeof__', '__str__', '__subclasshook__', 'id', 'last_update_string', 'n_value', 'n_value_string', 'name', 's_value', 'sub_type', 'switch_type', 'type']
2019-03-04 19:39:14.854  DIR(dir(DEVS.changed_device)): ['Describe', '__class__', '__delattr__', '__dir__', '__doc__', '__eq__', '__format__', '__ge__', '__getattribute__', '__gt__', '__hash__', '__init__', '__init_subclass__', '__le__', '__lt__', '__ne__', '__new__', '__reduce__', '__reduce_ex__', '__repr__', '__setattr__', '__sizeof__', '__str__', '__subclasshook__', 'id', 'last_update_string', 'n_value', 'n_value_string', 'name', 's_value', 'sub_type', 'switch_type', 'type']
domoticz: DIR(DEVS): ['Command', 'Devices', 'Log', '__doc__', '__loader__', '__name__', '__package__', '__spec__', 'changed_device', 'changed_device_name', 'is_daytime', 'is_nighttime', 'minutes_since_midnight', 'sunrise_in_minutes', 'sunset_in_minutes', 'user_variables']
2019-03-04 19:39:14.855  DIR(DEVS): ['Command', 'Devices', 'Log', '__doc__', '__loader__', '__name__', '__package__', '__spec__', 'changed_device', 'changed_device_name', 'is_daytime', 'is_nighttime', 'minutes_since_midnight', 'sunrise_in_minutes', 'sunset_in_minutes', 'user_variables']
domoticz: device: CPU_Usage Id: 1 n_value_string: 0.56 s_value: 0.56 Last updated: 2019-03-04 19:39:14 All typs:6 0 243
2019-03-04 19:39:14.857  device: CPU_Usage Id: 1 n_value_string: 0.56 s_value: 0.56 Last updated: 2019-03-04 19:39:14 All typs:6 0 243
domoticz: device: Memory Usage Id: 2 n_value_string: 19.47 s_value: 19.47 Last updated: 2019-03-04 19:38:44 All typs:6 0 243
2019-03-04 19:39:14.858  device: Memory Usage Id: 2 n_value_string: 19.47 s_value: 19.47 Last updated: 2019-03-04 19:38:44 All typs:6 0 243
domoticz: device: Process Usage Id: 3 n_value_string: 13.9300 s_value: 13.9300 Last updated: 2019-03-04 19:38:44 All typs:31 0 243
2019-03-04 19:39:14.859  device: Process Usage Id: 3 n_value_string: 13.9300 s_value: 13.9300 Last updated: 2019-03-04 19:38:44 All typs:31 0 243
domoticz: device: HDD /overlay Id: 5 n_value_string: 19.42 s_value: 19.42 Last updated: 2019-03-04 19:39:14 All typs:6 0 243
2019-03-04 19:39:14.860  device: HDD /overlay Id: 5 n_value_string: 19.42 s_value: 19.42 Last updated: 2019-03-04 19:39:14 All typs:6 0 243
domoticz: device: Podlipie_Pinger Id: 6 n_value_string: On s_value: 0 Last updated: 2019-03-03 09:55:08 All typs:0 0 17
2019-03-04 19:39:14.861  device: Podlipie_Pinger Id: 6 n_value_string: On s_value: 0 Last updated: 2019-03-03 09:55:08 All typs:0 0 17
domoticz: device: Ludynia Pinger Id: 7 n_value_string: On s_value: 0 Last updated: 2019-03-03 22:15:20 All typs:0 0 17
2019-03-04 19:39:14.862  device: Ludynia Pinger Id: 7 n_value_string: On s_value: 0 Last updated: 2019-03-03 22:15:20 All typs:0 0 17
domoticz: device: Glowackiego Pinger Id: 8 n_value_string: Off s_value: 0 Last updated: 2019-03-04 06:02:12 All typs:0 0 17
2019-03-04 19:39:14.863  device: Glowackiego Pinger Id: 8 n_value_string: Off s_value: 0 Last updated: 2019-03-04 06:02:12 All typs:0 0 17
domoticz: device: PiotrAP_Pinger Id: 9 n_value_string: On s_value: 0 Last updated: 2019-03-04 16:32:39 All typs:0 0 17
2019-03-04 19:39:14.864  device: PiotrAP_Pinger Id: 9 n_value_string: On s_value: 0 Last updated: 2019-03-04 16:32:39 All typs:0 0 17
domoticz: device: nBox Pinger Id: 10 n_value_string: On s_value: 0 Last updated: 2019-02-27 14:36:23 All typs:0 0 17
2019-03-04 19:39:14.865  device: nBox Pinger Id: 10 n_value_string: On s_value: 0 Last updated: 2019-02-27 14:36:23 All typs:0 0 17
domoticz: device: Glowackiego WG1 Pinger Id: 11 n_value_string: Off s_value: 0 Last updated: 2019-03-04 05:59:53 All typs:0 0 17
2019-03-04 19:39:14.866  device: Glowackiego WG1 Pinger Id: 11 n_value_string: Off s_value: 0 Last updated: 2019-03-04 05:59:53 All typs:0 0 17
domoticz: device: Przelacznik Id: 12 n_value_string: On s_value: 0 Last updated: 2019-03-04 19:14:08 All typs:73 0 244
2019-03-04 19:39:14.867  device: Przelacznik Id: 12 n_value_string: On s_value: 0 Last updated: 2019-03-04 19:14:08 All typs:73 0 244
domoticz: Python: Changed: ID: 1 Name: CPU_Usage, Type: 243, subType: 6, switchType: 0, s_value: 0.49, n_value: 0, n_value_string: 0.49, last_update_string: 2019-03-04 19:39:45
2019-03-04 19:39:45.417  Python: Changed: ID: 1 Name: CPU_Usage, Type: 243, subType: 6, switchType: 0, s_value: 0.49, n_value: 0, n_value_string: 0.49, last_update_string: 2019-03-04 19:39:45
domoticz: DEVS.changed_device_name: CPU_Usage
2019-03-04 19:39:45.418  DEVS.changed_device_name: CPU_Usage
domoticz: DEVS.changed_device.name: ID: 1 Name: CPU_Usage
2019-03-04 19:39:45.419  DEVS.changed_device.name: ID: 1 Name: CPU_Usage
domoticz: Python: Number of user_variables: 0
2019-03-04 19:39:45.419  Python: Number of user_variables: 0
domoticz: DIR(dir(DEVS.changed_device)): ['Describe', '__class__', '__delattr__', '__dir__', '__doc__', '__eq__', '__format__', '__ge__', '__getattribute__', '__gt__', '__hash__', '__init__', '__init_subclass__', '__le__', '__lt__', '__ne__', '__new__', '__reduce__', '__reduce_ex__', '__repr__', '__setattr__', '__sizeof__', '__str__', '__subclasshook__', 'id', 'last_update_string', 'n_value', 'n_value_string', 'name', 's_value', 'sub_type', 'switch_type', 'type']
2019-03-04 19:39:45.421  DIR(dir(DEVS.changed_device)): ['Describe', '__class__', '__delattr__', '__dir__', '__doc__', '__eq__', '__format__', '__ge__', '__getattribute__', '__gt__', '__hash__', '__init__', '__init_subclass__', '__le__', '__lt__', '__ne__', '__new__', '__reduce__', '__reduce_ex__', '__repr__', '__setattr__', '__sizeof__', '__str__', '__subclasshook__', 'id', 'last_update_string', 'n_value', 'n_value_string', 'name', 's_value', 'sub_type', 'switch_type', 'type']
domoticz: DIR(DEVS): ['Command', 'Devices', 'Log', '__doc__', '__loader__', '__name__', '__package__', '__spec__', 'changed_device', 'changed_device_name', 'is_daytime', 'is_nighttime', 'minutes_since_midnight', 'sunrise_in_minutes', 'sunset_in_minutes', 'user_variables']
2019-03-04 19:39:45.422  DIR(DEVS): ['Command', 'Devices', 'Log', '__doc__', '__loader__', '__name__', '__package__', '__spec__', 'changed_device', 'changed_device_name', 'is_daytime', 'is_nighttime', 'minutes_since_midnight', 'sunrise_in_minutes', 'sunset_in_minutes', 'user_variables']
domoticz: device: CPU_Usage Id: 1 n_value_string: 0.49 s_value: 0.49 Last updated: 2019-03-04 19:39:45 All typs:6 0 243
2019-03-04 19:39:45.423  device: CPU_Usage Id: 1 n_value_string: 0.49 s_value: 0.49 Last updated: 2019-03-04 19:39:45 All typs:6 0 243
domoticz: device: Memory Usage Id: 2 n_value_string: 19.47 s_value: 19.47 Last updated: 2019-03-04 19:38:44 All typs:6 0 243
2019-03-04 19:39:45.425  device: Memory Usage Id: 2 n_value_string: 19.47 s_value: 19.47 Last updated: 2019-03-04 19:38:44 All typs:6 0 243
domoticz: device: Process Usage Id: 3 n_value_string: 13.9300 s_value: 13.9300 Last updated: 2019-03-04 19:38:44 All typs:31 0 243
2019-03-04 19:39:45.426  device: Process Usage Id: 3 n_value_string: 13.9300 s_value: 13.9300 Last updated: 2019-03-04 19:38:44 All typs:31 0 243
domoticz: device: HDD /overlay Id: 5 n_value_string: 19.42 s_value: 19.42 Last updated: 2019-03-04 19:39:14 All typs:6 0 243
2019-03-04 19:39:45.427  device: HDD /overlay Id: 5 n_value_string: 19.42 s_value: 19.42 Last updated: 2019-03-04 19:39:14 All typs:6 0 243
domoticz: device: Podlipie_Pinger Id: 6 n_value_string: On s_value: 0 Last updated: 2019-03-03 09:55:08 All typs:0 0 17
2019-03-04 19:39:45.428  device: Podlipie_Pinger Id: 6 n_value_string: On s_value: 0 Last updated: 2019-03-03 09:55:08 All typs:0 0 17
domoticz: device: Ludynia Pinger Id: 7 n_value_string: On s_value: 0 Last updated: 2019-03-03 22:15:20 All typs:0 0 17
2019-03-04 19:39:45.429  device: Ludynia Pinger Id: 7 n_value_string: On s_value: 0 Last updated: 2019-03-03 22:15:20 All typs:0 0 17
domoticz: device: Glowackiego Pinger Id: 8 n_value_string: Off s_value: 0 Last updated: 2019-03-04 06:02:12 All typs:0 0 17
2019-03-04 19:39:45.430  device: Glowackiego Pinger Id: 8 n_value_string: Off s_value: 0 Last updated: 2019-03-04 06:02:12 All typs:0 0 17
domoticz: device: PiotrAP_Pinger Id: 9 n_value_string: On s_value: 0 Last updated: 2019-03-04 16:32:39 All typs:0 0 17
2019-03-04 19:39:45.431  device: PiotrAP_Pinger Id: 9 n_value_string: On s_value: 0 Last updated: 2019-03-04 16:32:39 All typs:0 0 17
domoticz: device: nBox Pinger Id: 10 n_value_string: On s_value: 0 Last updated: 2019-02-27 14:36:23 All typs:0 0 17
2019-03-04 19:39:45.432  device: nBox Pinger Id: 10 n_value_string: On s_value: 0 Last updated: 2019-02-27 14:36:23 All typs:0 0 17
domoticz: device: Glowackiego WG1 Pinger Id: 11 n_value_string: Off s_value: 0 Last updated: 2019-03-04 05:59:53 All typs:0 0 17
2019-03-04 19:39:45.433  device: Glowackiego WG1 Pinger Id: 11 n_value_string: Off s_value: 0 Last updated: 2019-03-04 05:59:53 All typs:0 0 17
domoticz: device: Przelacznik Id: 12 n_value_string: On s_value: 0 Last updated: 2019-03-04 19:14:08 All typs:73 0 244
2019-03-04 19:39:45.435  device: Przelacznik Id: 12 n_value_string: On s_value: 0 Last updated: 2019-03-04 19:14:08 All typs:73 0 244
domoticz: Python: Changed: ID: 2 Name: Memory Usage, Type: 243, subType: 6, switchType: 0, s_value: 19.47, n_value: 0, n_value_string: 19.47, last_update_string: 2019-03-04 19:40:05
2019-03-04 19:40:05.824  Python: Changed: ID: 2 Name: Memory Usage, Type: 243, subType: 6, switchType: 0, s_value: 19.47, n_value: 0, n_value_string: 19.47, last_update_string: 2019-03-04 19:40:05
domoticz: DEVS.changed_device_name: Memory Usage
2019-03-04 19:40:05.826  DEVS.changed_device_name: Memory Usage
domoticz: DEVS.changed_device.name: ID: 2 Name: Memory Usage
2019-03-04 19:40:05.827  DEVS.changed_device.name: ID: 2 Name: Memory Usage
domoticz: Python: Number of user_variables: 0
2019-03-04 19:40:05.828  Python: Number of user_variables: 0
domoticz: DIR(dir(DEVS.changed_device)): ['Describe', '__class__', '__delattr__', '__dir__', '__doc__', '__eq__', '__format__', '__ge__', '__getattribute__', '__gt__', '__hash__', '__init__', '__init_subclass__', '__le__', '__lt__', '__ne__', '__new__', '__reduce__', '__reduce_ex__', '__repr__', '__setattr__', '__sizeof__', '__str__', '__subclasshook__', 'id', 'last_update_string', 'n_value', 'n_value_string', 'name', 's_value', 'sub_type', 'switch_type', 'type']
2019-03-04 19:40:05.829  DIR(dir(DEVS.changed_device)): ['Describe', '__class__', '__delattr__', '__dir__', '__doc__', '__eq__', '__format__', '__ge__', '__getattribute__', '__gt__', '__hash__', '__init__', '__init_subclass__', '__le__', '__lt__', '__ne__', '__new__', '__reduce__', '__reduce_ex__', '__repr__', '__setattr__', '__sizeof__', '__str__', '__subclasshook__', 'id', 'last_update_string', 'n_value', 'n_value_string', 'name', 's_value', 'sub_type', 'switch_type', 'type']
domoticz: DIR(DEVS): ['Command', 'Devices', 'Log', '__doc__', '__loader__', '__name__', '__package__', '__spec__', 'changed_device', 'changed_device_name', 'is_daytime', 'is_nighttime', 'minutes_since_midnight', 'sunrise_in_minutes', 'sunset_in_minutes', 'user_variables']
2019-03-04 19:40:05.831  DIR(DEVS): ['Command', 'Devices', 'Log', '__doc__', '__loader__', '__name__', '__package__', '__spec__', 'changed_device', 'changed_device_name', 'is_daytime', 'is_nighttime', 'minutes_since_midnight', 'sunrise_in_minutes', 'sunset_in_minutes', 'user_variables']
domoticz: device: CPU_Usage Id: 1 n_value_string: 0.49 s_value: 0.49 Last updated: 2019-03-04 19:39:45 All typs:6 0 243
2019-03-04 19:40:05.832  device: CPU_Usage Id: 1 n_value_string: 0.49 s_value: 0.49 Last updated: 2019-03-04 19:39:45 All typs:6 0 243
domoticz: device: Memory Usage Id: 2 n_value_string: 19.47 s_value: 19.47 Last updated: 2019-03-04 19:40:05 All typs:6 0 243
2019-03-04 19:40:05.834  device: Memory Usage Id: 2 n_value_string: 19.47 s_value: 19.47 Last updated: 2019-03-04 19:40:05 All typs:6 0 243
domoticz: device: Process Usage Id: 3 n_value_string: 13.9300 s_value: 13.9300 Last updated: 2019-03-04 19:38:44 All typs:31 0 243
2019-03-04 19:40:05.835  device: Process Usage Id: 3 n_value_string: 13.9300 s_value: 13.9300 Last updated: 2019-03-04 19:38:44 All typs:31 0 243
domoticz: device: HDD /overlay Id: 5 n_value_string: 19.42 s_value: 19.42 Last updated: 2019-03-04 19:39:14 All typs:6 0 243
2019-03-04 19:40:05.837  device: HDD /overlay Id: 5 n_value_string: 19.42 s_value: 19.42 Last updated: 2019-03-04 19:39:14 All typs:6 0 243
domoticz: device: Podlipie_Pinger Id: 6 n_value_string: On s_value: 0 Last updated: 2019-03-03 09:55:08 All typs:0 0 17
2019-03-04 19:40:05.838  device: Podlipie_Pinger Id: 6 n_value_string: On s_value: 0 Last updated: 2019-03-03 09:55:08 All typs:0 0 17
domoticz: device: Ludynia Pinger Id: 7 n_value_string: On s_value: 0 Last updated: 2019-03-03 22:15:20 All typs:0 0 17
2019-03-04 19:40:05.839  device: Ludynia Pinger Id: 7 n_value_string: On s_value: 0 Last updated: 2019-03-03 22:15:20 All typs:0 0 17
domoticz: device: Glowackiego Pinger Id: 8 n_value_string: Off s_value: 0 Last updated: 2019-03-04 06:02:12 All typs:0 0 17
2019-03-04 19:40:05.841  device: Glowackiego Pinger Id: 8 n_value_string: Off s_value: 0 Last updated: 2019-03-04 06:02:12 All typs:0 0 17
domoticz: device: PiotrAP_Pinger Id: 9 n_value_string: On s_value: 0 Last updated: 2019-03-04 16:32:39 All typs:0 0 17
2019-03-04 19:40:05.842  device: PiotrAP_Pinger Id: 9 n_value_string: On s_value: 0 Last updated: 2019-03-04 16:32:39 All typs:0 0 17
domoticz: device: nBox Pinger Id: 10 n_value_string: On s_value: 0 Last updated: 2019-02-27 14:36:23 All typs:0 0 17
2019-03-04 19:40:05.843  device: nBox Pinger Id: 10 n_value_string: On s_value: 0 Last updated: 2019-02-27 14:36:23 All typs:0 0 17
domoticz: device: Glowackiego WG1 Pinger Id: 11 n_value_string: Off s_value: 0 Last updated: 2019-03-04 05:59:53 All typs:0 0 17
2019-03-04 19:40:05.845  device: Glowackiego WG1 Pinger Id: 11 n_value_string: Off s_value: 0 Last updated: 2019-03-04 05:59:53 All typs:0 0 17
domoticz: device: Przelacznik Id: 12 n_value_string: On s_value: 0 Last updated: 2019-03-04 19:14:08 All typs:73 0 244
2019-03-04 19:40:05.846  device: Przelacznik Id: 12 n_value_string: On s_value: 0 Last updated: 2019-03-04 19:14:08 All typs:73 0 244
domoticz: Python: Changed: ID: 3 Name: Process Usage, Type: 243, subType: 31, switchType: 0, s_value: 13.9600, n_value: 0, n_value_string: 13.9600, last_update_string: 2019-03-04 19:40:05

```

**Comment by gizmocuz** [comment]
Did you posted this on the forum ?

**Comment by kofec** [comment]
No. I think that it is rather an issue ? Is it expected behavior ? I did what you suggest:
https://www.domoticz.com/forum/viewtopic.php?f=65&t=27223
Thanks

**Comment by gizmocuz** [comment]
Yes, best to ask on the forum first, it could be a script issue

**Comment by kofec** [comment]
OK let me clarify here. I read wiki again -  Regarding to 
https://www.domoticz.com/wiki/Events#By_adding_your_own_scripts_to_the_scripts.2Flua.2F_dir_in_your_domoticz_installation_folder.

This is only for lua script right ? For python it is not implemented ? - maybe I can and this note on this link if you confirm it ?
I mean I have one device "przelacznik" and two scripts:
../lua/script_device_przelacznik.lua
../python/script_device_przelacznik.py

And **python** script is triggered by **any** event but **lua** script is triggered **only** by events from "przelacznik" device. So the below description is only for lua right:

> Note: As of version V3.8050 device scripts can be named the same as the device to only evaluate the Lua code when this device changes. Eg. if you use script_device_switch_1.lua and you have a device named 'Switch 1', this script will only trigger when this device updates. You should only use lowercase characters in the file name for this to work. Spaces are set with an underscore (_). If you name the script randomly, eg. script_device_somethingelse.lua, and there is no such device named 'somethingelse', it will evaluate the script on all device changes, the same behavior as pre V3.8050.

**Comment by moroen** [comment]
Yes, python and Lua handles this differently.

Python scripts with the name script_device_*.py gets triggered by a change in any device. This is the intended behaviour, partly done to enable one script to react to more than one device. The scripts should do some kind of test, for instance checking the changed_device obejct, to make sure that only the correct device(s) have changed and then perform the rest of the script. 

Back when python event script system was updated to use python3, I looked at implementing this the same way as the lua events, but decided against it. 

If the need for having a script only being triggered by a device with the same name as the script is high, I'll could probably find the time to implement it. Or as a work around, it's possible to create a script_device_master.py that checks the changed_device object, check if there is a script matching the device and execute it.
